core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -170,11 +174,11 @@ public void prepareRun(BatchJoinerContext context) {
   @Override
   public void initialize(BatchJoinerRuntimeContext context) {
     FailureCollector collector = context.getFailureCollector();
-    init(context.getInputSchemas(), collector);
-    collector.getOrThrowException();
-
     Map<String, Schema> inputSchemas = context.getInputSchemas();
-    outputSchema = context.getOutputSchema();
+    init(inputSchemas, collector);
+    collector.getOrThrowException();
```

Comment:
do we need this?

Reply:
It's needed to handle validation failures in join key validation failures.